File: backend/fastapi/app/api/v1/health.py
"""
健康检查和监控API路由
"""

from datetime import datetime
from fastapi import APIRouter, HTTPException

from app.models.responses import HealthResponse, MetricsResponse
from app.core.cache import cache_manager
from app.services.golang_service import golang_service
from app.config import settings

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
async def health_check():
    """健康检查接口"""
    # 检查Redis连接
    redis_status = "healthy" if await cache_manager.ping() else "unhealthy"
    
    # 检查Golang服务连接
    golang_status = "healthy" if await golang_service.health_check() else "unhealthy"
    
    overall_status = "healthy" if redis_status == "healthy" and golang_status == "healthy" else "degraded"
    
    return HealthResponse(
        status=overall_status,
        services={
            "redis": redis_status,
            "golang": golang_status
        },
        timestamp=datetime.now().isoformat(),
        version=settings.APP_VERSION
    )


@router.get("/metrics", response_model=MetricsResponse)
async def get_metrics():
    """系统指标接口"""
    try:
        # 获取一些基础指标
        redis_info = await cache_manager.get_info("memory")
        
        return MetricsResponse(
            redis_memory_used=redis_info.get("used_memory_human", "unknown"),
            active_connections="unknown",  # 实际应该从连接池获取
            cache_hit_rate="unknown",      # 实际应该计算缓存命中率
            timestamp=datetime.now().isoformat()
        )
    except Exception as e:
        raise HTTPException(status_code=503, detail=f"获取指标失败: {e}") 

File: backend/hono/src/routes/health.ts
/**
 * 健康检查和监控API路由
 */

import { Hono } from 'hono';
import { cacheManager } from '../core/cache';
import { golangService } from '../services/golang-service';
import { settings } from '../config';
import { HealthResponse, MetricsResponse } from '../types';

const health = new Hono();

// 健康检查接口
health.get('/health', async (c) => {
  // 检查Redis连接
  const redisStatus = (await cacheManager.ping()) ? 'healthy' : 'unhealthy';

  // 检查Golang服务连接
  const golangStatus = (await golangService.healthCheck()) ? 'healthy' : 'unhealthy';

  const overallStatus = redisStatus === 'healthy' && golangStatus === 'healthy' ? 'healthy' : 'degraded';

  const response: HealthResponse = {
    status: overallStatus,
    services: {
      redis: redisStatus,
      golang: golangStatus,
    },
    timestamp: new Date().toISOString(),
    version: settings.app.version,
  };

  return c.json(response);
});

// 系统指标接口
health.get('/metrics', async (c) => {
  try {
    // 获取一些基础指标
    const redisInfo = await cacheManager.getInfo('memory');

    const response: MetricsResponse = {
      redis_memory_used: redisInfo.used_memory_human || 'unknown',
      active_connections: 'unknown', // 实际应该从连接池获取
      cache_hit_rate: 'unknown', // 实际应该计算缓存命中率
      timestamp: new Date().toISOString(),
    };

    return c.json(response);
  } catch (error) {
    console.error('获取指标失败:', error);
    return c.json({ error: '获取指标失败' }, 503);
  }
});

export default health; 